### ISSUE TITLE
Unable to build jaxlib on Windows10

### ISSUE THREAD

**Issue by keepingupwithdeeplearning** [google/jax#7131]
Building jaxlib from source on Windows10 using bazel 4.0.0 and msys2 keeps giving me the following cuda related error:

![Screenshot (54)](https://user-images.githubusercontent.com/32151860/123731823-4c555e80-d8dc-11eb-81a7-26cf3bb4ab22.png)

I used the following command to build:

python .\build\build.py --enable_cuda --cuda_path="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1" --cudnn_path="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1" --cuda_compute_capabilities="6.1" --cuda_version="10.1" --cudnn_version="7.6.5"

Building using bazel 4.1.0 gives a cpu related error:
![Screenshot (55)](https://user-images.githubusercontent.com/32151860/123732002-95a5ae00-d8dc-11eb-90fa-298136c2a6f7.png)

**Comment by jakevdp** [comment]
Thanks for the question – Windows builds are not very well supported at the moment. In case you haven't seen them, it may be worth looking through the replies on a few previous issues related to this topic: #7079 #6637 #5985

**Comment by skye** [comment]
FYI @zhangqiaorjc. The jaxlib build has been broken on Windows for a while now unfortunately, see https://github.com/google/jax/issues/7079. I'm gonna close this issue; please follow #7079, and comment there if you discover more info regarding the failure.

**Comment by GMCobraz** [comment]
Hi @keepingupwithdeeplearning,

May I know how how you manage to run those commands in anaconda prompt?
I installed bazel v5 and also mysys2.
Before that, I cannot install realpath using mysys2, did you face this issues?

When I try to run the commands,
python .\build\build.py `
  --enable_cuda `
  --cuda_path='C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1' `
  --cudnn_path='C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1' `
  --cuda_compute_capabilities='5.0' `
  --cuda_version='10.1' `
  --cudnn_version='7.6.5'

I got this error.
python: can't open file '.\build\build.py': [Errno 2] No such file or directory.

Please show or point out my error if possible.

Thank you very much

**Comment by keepingupwithdeeplearning** [comment]
Hey, 
You don't need to install realpath separately. It's in mysys2 bin folder. Just make sure you set the path to mysys2 bin folder. I had issues with bazel v5 (they have removed couple of functions from the newer versions). So I tried with bazel v4.0 and bazel 4.1.

Hope this helps. Please do let me know if you manage to build jaxlib. The newer version of jaxlib doesn't help either.

**Comment by keepingupwithdeeplearning** [comment]
The path should be C:\Users\Aries\msys64\usr\bin

**Comment by GMCobraz** [comment]
Hi @keepingupwithdeeplearning ,
Even I set the path, I still get the same error, mentioned that no such file or directory.
Please help

Thank you very much.

**Comment by keepingupwithdeeplearning** [comment]
Save bazel.exe in a folder  (eg: bazel_folder) and set the path to the folder:

C:\Users\Aries\bazel_folder

**Comment by GMCobraz** [comment]
Hi @keepingupwithdeeplearning,

Thanks for the replied. 
I did run the jax but still failed to build the jaxlib. I faced similar errors as yours.

I saw in post #7079, someone claimed that they are succefully build the jaxlib with python 3.8, cuda v11.2, cudnn v8.1.

I suspect this is due to the gcc version from msys2.
If we check the gcc version from msys2 or cmd. We found out that the gcc version show is v10.

I tried install cuda in ubuntu too, with cuda v10.1, cudnn v7.6.5.
I remember the gcc and g++ version must be v8. 
It will have error when gcc & g++ version larger than 8.

So I suspect this is the issue if someone try to build jaxlib with cuda v10.1, cudnn v7.6.5 will fail due to incompatible gcc & g++ version.
Maybe cuda v11.2, cudnn v8.1 are running fine with gcc & g++ version 10. 
This is my hypothesis only. 

Thank you very much.
